inc/class_datacache.php

```diff
@@ -134,7 +134,7 @@ function cache()
 			$query = $db->simple_select("datacache", "title,cache");
 			while($data = $db->fetch_array($query))
 			{
-				$this->cache[$data['title']] = unserialize($data['cache']);
+				$this->cache[$data['title']] = my_unserialize($data['cache']);
```

Comment:
This call would very probably impact performance of large forums with many categories/forums, and so does the following call `$data = my_unserialize($cache_data['cache']);` in line 220 when the `forums` cache is read.

Reply:
Yes. Contents of most entries could be large strings, and that'd potentially cause performance issue.